ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala

```diff
@@ -69,106 +70,72 @@ class Endpoints(
   import encoder.implicits._
   import json.JsonProtocol._
   import util.ErrorOps._
-  import Uri.Path._
+
+  // Weird signature required, because it looks like Scala
+  // does not see a difference between Future[HttpResponse] & other inner types.
+  // Without it, it will complain that the function has been defined twice.
+  @nowarn("msg=parameter value ev in method toRoute is never used")
+  def toRoute[A <: HttpResponse](res: Future[A])(implicit
+      ev: A =:= HttpResponse
+  ): Route = _ => res map Complete
```

Comment:
```suggestion
  def toRoute(res: Future[HttpResponse]): Route = _ => res map Complete
```
this compiles just fine for me. What am I missing?

Reply:
Alternatively, maybe consider just giving this function a different name instead of doing some weird overloading.

Reply:
ok let me try locally, maybe I did mess something up.

Reply:
I was thinking of a simple manual test, add some `println`s in the futures and see if they get triggered for routes that aren’t hit.

Reply:
ok weird, you're right. Will change that stuff (as I just noticed that I cannot press the button here anymore after committing the other suggestions :sweat_smile:).

Reply:
many thanks, done with 2f55802